algorithms/src/Kokkos_Sort.hpp

```diff
@@ -308,18 +315,32 @@ class BinSort {
                   values_range_begin - range_begin);
 
       parallel_for("Kokkos::Sort::CopyPermute",
-                   Kokkos::RangePolicy<execution_space>(0, len), functor);
+                   Kokkos::RangePolicy<ExecutionSpace>(exec, 0, len), functor);
     }
 
     {
       copy_functor<ValuesViewType, scratch_view_type> functor(
           values, range_begin, sorted_values);
 
       parallel_for("Kokkos::Sort::Copy",
-                   Kokkos::RangePolicy<execution_space>(0, len), functor);
+                   Kokkos::RangePolicy<ExecutionSpace>(exec, 0, len), functor);
     }
 
-    execution_space().fence("Kokkos::Sort: fence after sorting");
+    exec.fence("Kokkos::Sort: fence after sorting");
+  }
+
+  // Sort a subset of a view with respect to the first dimension using the
+  // permutation array
+  template <class ValuesViewType>
+  void sort(ValuesViewType const& values, int values_range_begin,
+            int values_range_end) const {
+    sort(typename ValuesViewType::execution_space{}, values, values_range_begin,
+         values_range_end);
+  }
+
+  template <class ExecutionSpace, class ValuesViewType>
+  void sort(ExecutionSpace const& exec, ValuesViewType const& values) const {
+    this->sort(exec, values, 0, /*values.extent(0)*/ range_end - range_begin);
   }
```

Comment:
Adding a new overload?  `sort(ValuesViewType)` does not exist does it?

Reply:
Yes, it did, see https://github.com/kokkos/kokkos/blob/f5c5b1d7fb58336c86b7fb3b408a09ff93cd57b1/algorithms/src/Kokkos_Sort.hpp#L325-L328.